src/ws/cockpitcertificate.c

```diff
@@ -133,10 +133,10 @@ openssl_make_dummy_cert (const gchar *key_file,
 {
   gboolean ret = FALSE;
   gint exit_status;
-  gchar *stderr_str = NULL;
-  gchar *command_line = NULL;
-  gchar *ssl_config = NULL;
-  gchar *subject = generate_subject ();
+  g_autofree gchar *stderr_str = NULL;
+  g_autofree gchar *command_line = NULL;
+  g_autofree gchar *ssl_config = NULL;
+  g_autofree gchar *subject = generate_subject ();
 
   /* make config file with subjectAltName for localhost and our tests */
   ssl_config = create_temp_file (g_get_tmp_dir (), "ssl.conf.XXXXXX", error);
```

Comment:
Not your bug, but worth fixing while you're in here: the `return FALSE` in the event of fail from `g_file_set_contents()` will leave the (empty) temp file dangling.

While you're at it, change the `if (!ssl_config)` to a `goto out;` for consistency.

Reply:
Well spotted, thanks! Fixed in a separate commit.